eng/WpfArcadeSdk/tools/configure-helix-machine.ps1

```diff
@@ -0,0 +1,30 @@
+# This file prepares the helix machine for our tests runs
+$dotnetLocation = Join-Path (Split-Path -Parent $script:MyInvocation.MyCommand.Path) "dotnet"
+
+# Set DOTNET_ROOT variables so the host can find it
+Set-Item -Path "env:DOTNET_ROOT(x86)" -Value $dotnetLocation
+Set-Item -Path "env:DOTNET_ROOT" -Value $dotnetLocation
+
+$runtimes = dotnet --list-runtimes
+foreach ($rt in $runtimes)
+{
+    if ($rt.StartsWith("Microsoft.WindowsDesktop.App"))
+    {
+        $version = $rt.Split(" ")[1]
+    }
+}
+
+# Rewrite the *.runtimeconfig.json files to match the version of the runtime on the machine
```

Comment:
Maybe I am misunderstanding this, but don't we dictate the version of the runtime on the Helix machine?  Why would we need to re-write it?  Multi-framework should mean that our deps.json is built targeting the chosen framework version from global,.json, right?  So we just need to ensure we have that framework on the machine.  What am I missing?

Reply:
the infrastructure isn't built out in the open, so i was getting errors about not being able to find the proper version of WindowsDesktop.App. This should only be a temporary thing until we move it out into the open.

Reply:
actually i'm now wondering if this only worked because the helix machine had some old install of the WindowsDesktop runtime on it. now that i've updated the DotNetCliVersion to use the version of NetCoreApp that we compile WPF against, this doesn't work. We're going to publish the test infrastructure package as self-contained, so we can minimize these dependencies. once it's open-sourced we can compile it like we do today and things will just work

Reply:
The infrastructure isn't WindowsDesktop, afaik.  @miguep, nothing in QV or STI has anything to do with WindowsDesktop, correct?

Reply:
Was the infra package built during the time period where Arcade had the version bug?

Reply:
Publishing as self contained doesn't test what we want to test, unless you also copy over the entirety of the dotnet called out in multi-framework.  Also how does SCA infra work with a build -test run, what are we testing in that instance?

Reply:
QV has a dependency on WinForms for displaying the progress in the bottom right. We should be able to depend on the private WinForms package once we move it out into the open. Otherwise, I'd vote for removing the dependency entirely so it can just build against the `Microsoft.NET.Sdk`. What does SCA mean?

Which version bug are you referring to? I don't think we technically care about what QV or STI run against, as long as the actual tests themselves don't run against it. Unfortunately, building the infra as `SelfContained` breaks the `build -test` scenario (which seems weird to me). The test executables are now no longer able to find the version of `Microsoft.NETCore.App` that gets installed by Arcade in the `.dotnet` folder. I'm not exactly sure why this is, I would think the self contained QV, STI, and DrtXaml would still inherit the proper environment variables that point to the local install. There must be something here about a self-contained process calling `Process.Start` on a non-self-contained app that breaks this scenario. Maybe @ericstj can share more info about this.